Set-DomainUser.ps1

```diff
@@ -1,720 +1,1124 @@
+    & "$hardeningScriptRoot\PSMHardening.ps1"
+    Set-Location $CurrentLocation
+}
+
+Function Invoke-PSMConfigureAppLocker {
+    <#
+    .SYNOPSIS
+    Runs the AppLocker PowerShell script
+    .DESCRIPTION
+    Runs the AppLocker PowerShell script
+    .PARAMETER psmRootInstallLocation
+    PSM root installation folder
+    #>
+    param (
+        [Parameter(Mandatory = $true)]
+        $psmRootInstallLocation
+    ) 
+    Write-Verbose "Starting PSMConfigureAppLocker"
+    $hardeningScriptRoot = "$psmRootInstallLocation\Hardening"
+    $CurrentLocation = Get-Location
+    Set-Location $hardeningScriptRoot
+    & "$hardeningScriptRoot\PSMConfigureAppLocker.ps1"
+    Set-Location $CurrentLocation 
+}
+
+Function New-VaultAdminObject {
+    <#
+    .SYNOPSIS
+    Onboards an account in the vault
+    .DESCRIPTION
+    Onboards an account in the vault
+    .PARAMETER pvwaAddress
+    Address of the PVWA
+    .PARAMETER pvwaToken
+    Token to log into PVWA using APIs
+    .PARAMETER name
+    Name of the account (PSMConnect/PSMAdminConnect)
+    .PARAMETER domain
+    Domain of the users needed to be onboarded
+    .PARAMETER Credentials
+    Credentials to be onboarded (has both the username and password)     
+    .PARAMETER platformID
+    The Platform to onboard the account to. We will use the PlatformID in this script from what we create.
+    #>
+    param (
+        [Parameter(Mandatory = $true)]
+        $pvwaAddress,
+        [Parameter(Mandatory = $true)]
+        $pvwaToken,
+        [Parameter(Mandatory = $true)]
+        $name,
+        [Parameter(Mandatory = $true)]
+        [String]$domain,
+        [Parameter(Mandatory = $true)]
+        [PSCredential]$Credentials,
+        [Parameter(Mandatory = $true)]
+        $platformID,
+        [Parameter(Mandatory = $false)]
+        $safe = "PSM"
+    ) 
+    
+    $username = $Credentials.username.Replace('\', '')
+    $password = $Credentials.GetNetworkCredential().password
+    $body = @{
+        name                      = $name
+        address                   = $domain
+        userName                  = $username
+        safeName                  = $safe
+        secretType                = "password"
+        secret                    = $password
+        platformID                = $platformID
+        platformAccountProperties = @{"LogonDomain" = $domain }
+    }
+    $url = $pvwaAddress + "/PasswordVault/api/Accounts"
+    $json = $body | ConvertTo-Json
+    try {
+        $result = Invoke-RestMethod -Method POST -Uri $url -Body $json -Headers @{ "Authorization" = $pvwaToken } -ContentType "application/json" -ErrorVariable ResultError
+        return $result
+    }
+    catch {
+        try {
+            $ErrorMessage = $ResultError.Message | ConvertFrom-Json
+            return $ErrorMessage
+        }
+        catch {
+            Write-Error ("Error creating user: {0}" -f $ResultError.Message)
+            exit 1
+        }
+    }
+}
+
+Function Add-AdminUserToTS {
```

Comment:
Several changes to this process.
1. Broke out the 2 commands into separate functions
2. Use Powershell to invoke WMI instead of cmd so it doesn't have to invoke a separate shell and is better able to catch errors